src/core/datastructures/tfprimitiveset.cpp

```diff
@@ -235,10 +252,15 @@ void TFPrimitiveSet::onTFPrimitiveChange(const TFPrimitive* p) {
 }
 
 void TFPrimitiveSet::serialize(Serializer& s) const {
+    type_.serialize(s, PropertySerializationMode::All);
```

Comment:
Why all?

Reply:
in order to see the type of the TF at all times.